src/machine.rs

```diff
@@ -110,4 +114,45 @@ impl Machine {
         let machine_file = MachineFile::from(machine_file_response.data);
         Ok(machine_file)
     }
+
+    pub async fn ping(&self) -> Result<(), Error> {
+        let client: Client = Client::default();
+        let _response: Response<MachineResponse> = client
+            .post(&format!(
+                "machines/{}/actions/ping", self.id),
+                None::<&()>,
+                None::<&()>
+            )
+            .await?;
+        Ok(())
+    }
+
+
+    pub fn monitor(self: Arc<Self>, heartbeat_interval: Duration, tx: Option<Sender<Error>>) -> BoxFuture<'static, ()> {
+        
```

Comment:
Reference https://github.com/keygen-sh/keygen-go/blob/master/machine.go#L147, I think we should immediately call the `ping` method to send a heartbeat at first.

```rust
pub fn monitor(
    self: Arc<Self>,
    heartbeat_interval: Duration,
    tx: Option<Sender<Error>>,
) -> BoxFuture<'static, ()> {
    async move {
        if let Err(e) = self.ping().await {
            match &tx {
                Some(tx) => {
                    tx.send(e).unwrap();
                }
                _ => {
                    log::error!("Heartbeat monitor failed, aborting.");
                }
            };
            return;
        }

        ...
  }

```